### PR TITLE
#1786 :- Replace DocumentToAssetTest unit test with an integration test

### LINKED ISSUE #1786: Replace DocumentToAssetTest unit test with an integration test
Replace `\Magento\AdobeStockImage\Test\Unit\Model\Extract\DocumentToAssetTest` unit test with an integration test

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR will replace the Unit test to Integration test for `\Magento\AdobeStockImage\Test\Unit\Model\Extract\DocumentToAssetTest`
### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration#1786: Replace DocumentToAssetTest unit test with an integration test

### CODE REVIEW

**Comment by konarshankar07** [comment]
@magento run all tests

File: AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        array $categoryData,
+        array $creatorData,
+        Document $document,
+        array $additionalData
+    ): void {
+        $this->documentToAsset = Bootstrap::getObjectManager()->get(
+            AdobeStockAsset::class
+        );
+        $documentToAsset = $this->documentToAsset->convert($document, $additionalData);
+        $this->assertEquals($creatorData, $documentToAsset->getCreator()->getData());
```
Comment (sivaschenko) [inline AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php]: `\Magento\AdobeStockAssetApi\Api\Data\CreatorInterface` and other data interfaces does not have the `getData` method declared.

Can you please replace `getData` calls with `\Magento\Framework\Reflection\DataObjectProcessor::buildOutputDataArray`

File: AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+     * @param array $additionalData
+     */
+    public function testConvert(
+        array $assetData,
+        array $categoryData,
+        array $creatorData,
+        Document $document,
+        array $additionalData
+    ): void {
+        $this->documentToAsset = Bootstrap::getObjectManager()->get(
```
Comment (sivaschenko) [inline AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php]: The `documentToAsset` instance is created for each test run in `setUp` method, no need to create it in the test

File: AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            AdobeStockAsset::class
+        );
+        $documentToAsset = $this->documentToAsset->convert($document, $additionalData);
```
Comment (sivaschenko) [inline AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php]: Naming clarification suggestion
```suggestion
        $asset = $this->documentToAsset->convert($document, $additionalData);
```

File: AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                        ],
+                        'creator_name' => 'Creator',
+                        'creator_id' => 3,
+                        'is_licensed' => 1
+                    ]
+                ),
+                [
+                    'media_gallery_id' => 5
+                ]
+            ]
```
Comment (sivaschenko) [inline AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php]: I'd create the document instance in  the test, not to duplicate the data. I believe it's enough to have only two  parameters in the test
```suggestion
            'case1' => [
                'data' => [
                        'id' => 1,
                        'category' => [
                            'id' => 2,
                            'name' => 'The Category'
                        ],
                        'creator_name' => 'Creator',
                        'creator_id' => 3,
                        'is_licensed' => 1
                    ],
                'additionaData' => [
                    'media_gallery_id' => 5
                ]
            ]
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @konarshankar07 ! Please see my review comments

**Comment by konarshankar07** [comment]
@magento run all tests

**Comment by konarshankar07** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**